pkg/controller/ps/controller.go

```diff
@@ -537,15 +537,22 @@ func (r *PerconaServerMySQLReconciler) reconcileMySQLConfiguration(
 		return "", errors.Wrapf(err, "get ConfigMap/%s", cmName)
 	}
 
-	// Cleanup if user removed the configuration from CR
 	if cr.Spec.MySQL.Configuration == "" {
 		exists, err := k8s.ObjectExists(ctx, r.Client, nn, currCm)
 		if err != nil {
 			return "", errors.Wrapf(err, "check if ConfigMap/%s exists", cmName)
 		}
 
 		if !exists || !metav1.IsControlledBy(currCm, cr) {
-			return "", nil
+			//ConfigMap exists and is created by the user
```

Comment:
if we're here it doesn't mean "configmap exists and is created by the user". it means cm doesn't exist or it's created by user. do we want to calculate the hash if it doesn't exist?

Reply:
Yes, the comment is misleading (or plainly wrong), if `currCM` does not exist we will return an empty string since `currCM.Data` is empty` and that's why this works correctly. I'll update the impl so it is more clear by having additional checks outside this one, like `if exists && !metav1.IsControlledBy(currCm, cr) {...}`.

Reply:
[0c79c50](https://github.com/percona/percona-server-mysql-operator/pull/381/commits/0c79c50f7995943f3e21439e5d3712df04a50184)